internal/core/services/worker/worker_test.go

```diff
@@ -0,0 +1,175 @@
+//+build unit
+
+package worker
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	opflow "github.com/topfreegames/maestro/internal/adapters/operation_flow/mock"
+	opstorage "github.com/topfreegames/maestro/internal/adapters/operation_storage/mock"
+	"github.com/topfreegames/maestro/internal/core/entities/operation"
+	"github.com/topfreegames/maestro/internal/core/operations"
+	mockoperation "github.com/topfreegames/maestro/internal/core/operations/mock"
+	"github.com/topfreegames/maestro/internal/core/services/operations_registry"
+)
+
+func TestSchedulerOperationsExecutionLoop(t *testing.T) {
+	t.Run("successfully runs a single operation", func(t *testing.T) {
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		operationFlow := opflow.NewMockOperationFlow(mockCtrl)
+		operationStorage := opstorage.NewMockOperationStorage(mockCtrl)
+
+		operationName := "test_operation"
+		operationDefinition := mockoperation.NewMockDefinition(mockCtrl)
+		operationExecutor := mockoperation.NewMockExecutor(mockCtrl)
+		operationExecutor.EXPECT().Name().Return(operationName).AnyTimes()
+		operationDefinition.EXPECT().Name().Return(operationName).AnyTimes()
+
+		defFunc := func() operations.Definition { return operationDefinition }
+		registry := operations_registry.NewRegistry()
+		registry.Register(operationName, defFunc)
+
+		workerService := NewWorker(&WorkerOptions{operationStorage, operationFlow, []operations.Executor{operationExecutor}, &registry})
+
+		expectedOperation := &operation.Operation{
+			ID:             "random-operation-id",
+			SchedulerName:  "random-scheduler",
+			Status:         operation.StatusPending,
+			DefinitionName: operationName,
+		}
+
+		operationDefinition.EXPECT().Unmarshal(gomock.Any()).Return(nil)
+		operationDefinition.EXPECT().ShouldExecute(context.Background(), []*operation.Operation{}).Return(true)
+		operationExecutor.EXPECT().Execute(context.Background(), expectedOperation, operationDefinition).Return(nil)
+
+		operationFlow.EXPECT().NextOperationID(context.Background(), expectedOperation.SchedulerName).Return(expectedOperation.ID, nil)
+		operationStorage.EXPECT().GetOperation(context.Background(), expectedOperation.SchedulerName, expectedOperation.ID).Return(expectedOperation, []byte{}, nil)
+		operationStorage.EXPECT().UpdateOperationStatus(context.Background(), expectedOperation.SchedulerName, expectedOperation.ID, operation.StatusInProgress).Return(nil)
+		operationStorage.EXPECT().UpdateOperationStatus(context.Background(), expectedOperation.SchedulerName, expectedOperation.ID, operation.StatusFinished).Return(nil)
+		operationFlow.EXPECT().NextOperationID(context.Background(), expectedOperation.SchedulerName).Return("", context.Canceled)
```

Comment:
Observing the error as context.Canceled turns to me little bit harder to understand the stop criteria, maybe adding some comments would help us to gain more legibility.

Reply:
I added a comment on the tests to make it more clear.